networkx/algorithms/shortest_paths/astar.py

```diff
@@ -67,6 +67,14 @@ def astar_path(G, source, target, heuristic=None, weight="weight"):
     >>> print(nx.astar_path(G, (0, 0), (2, 2), heuristic=dist, weight="cost"))
     [(0, 0), (0, 1), (0, 2), (1, 2), (2, 2)]
 
+    Notes
+    -----
+    Edge weight attributes must be numerical.
+    Distances are calculated as sums of weighted edges traversed.
```

Comment:
This was taken from the Dijkstra docstring which is a good starting point, but this also includes the heuristic (when supplied), right?

Reply:
Good point @rossbar ,
We should explain whether the heuristic function can also return None or not. I think it should not return None (to avoid a TypeError when adding None to a number), but I'm not sure.

Reply:
I agree that the heuristic function should never return None and fortunately the docstring already stresses this point. Here's the first few lines of the description for the `heuristic` parameter:

> A function to evaluate the estimate of the distance
> from the a node to the target.  The function takes
> two nodes arguments and must return a number.

Reply:
e775fa992dd51d8ff413e3133a667bbe7330b2c4 updates the description for the `weight` parameter to say "must return a number **or None**", for both A* and Dijkstra.

Reply:
Thanks for this!
Should the description for the `weight` parameter say what `None` stands for?  Perhaps:
"must return a number or return None to indicate a hidden edge."